app/models/profile.rb

```diff
@@ -0,0 +1,14 @@
+class Profile < ApplicationRecord
+  belongs_to :user
+
+  # This method generates typed accessors for all active profile fields
+  def self.define_store_accessors!
+    ProfileField.active.find_each do |field|
+      store_attribute :data, field.attribute_name, field.type
+    end
+  end
+
+  define_store_accessors!
+
+  validates :data, presence: true
```

Comment:
The column is non-nullable with a default value of `{}`, which normally greatly simplifies working with JSONB data.